tests/forcefields/test_jobs.py

```diff
@@ -232,6 +248,13 @@ def test_mace_relax_maker(
         assert output1.output.energy == approx(-0.051912, rel=1e-4)
         assert output1.output.n_steps == 4
 
+    # get space group number of input structure
+    _, final_spg_num = output1.output.structure.get_space_group_info()
+    if fix_symmetry:
+        assert init_spg_num == final_spg_num == 12
```

Comment:
@JonathanSchmidt1 this test is failing which seems to suggest `fix_symmetry` doesn't work for MACE? maybe i'm missing something?

Reply:
Just checked on my phone so I am sorry if I overlooked sth but it looks to me like the energy checks are failing before right now. Two ideas concerning this:
When Fixing the symmetry the structure is first refined and second you changed the fmax both might lead to a different energy.
If the symmetry check is failing later on which I cannot see on the phone, it might be because we are not determining the initial symmetry with the same symprec  in the si-test as the one used for fix_symmetry. I also think the angle tolerance in pymatgen might not be the default one used in ase ( 5 vs -1 I believe).

 Gonna take a look later in the evening to see if I can find the problem.

> =========================== short test summary info ============================
> FAILED tests/forcefields/test_jobs.py::test_mace_relax_makera[/home/runner/work/atomate2/atomate2/tests/test_data/forcefields/mace/MACE.model-True-True-0.01] - assert -0.07111746072769165 == -0.0526856 ± 5.3e-03
>   
>   comparison failed
>   Obtained: -0.07111746072769165
>   Expected: -0.0526856 ± 5.3e-03
> FAILED tests/forcefields/test_jobs.py::test_mace_relax_makera[/home/runner/work/atomate2/atomate2/tests/test_data/forcefields/mace/MACE.model-True-False-0.01] - assert -0.0711173564195633 == -0.0526856 ± 5.3e-03
>   
>   comparison failed
>   Obtained: -0.0711173564195633
>   Expected: -0.0526856 ± 5.3e-03
> FAILED tests/forcefields/test_jobs.py::test_mace_relax_makera[/home/runner/work/atomate2/atomate2/tests/test_data/forcefields/mace/MACE.model-True-True-0.1] - assert -0.07111746072769165 == -0.0526856 ± 5.3e-03
>   
>   comparison failed
>   Obtained: -0.07111746072769165
>   Expected: -0.0526856 ± 5.3e-03
> FAILED tests/forcefields/test_jobs.py::test_mace_relax_makera[/home/runner/work/atomate2/atomate2/tests/test_data/forcefields/mace/MACE.model-False-True-0.01] - assert -0.06772968173027039 == -0.051912 ± 5.2e-06
>   
>   comparison failed
>   Obtained: -0.06772968173027039
>   Expected: -0.051912 ± 5.2e-06
> FAILED tests/forcefields/test_jobs.py::test_mace_relax_makera[/home/runner/work/atomate2/atomate2/tests/test_data/forcefields/mace/MACE.model-False-False-0.01] - assert -0.06772962212562[561](https://github.com/materialsproject/atomate2/actions/runs/8586508423/job/23529051664?pr=789#step:6:562) == -0.051912 ± 5.2e-06
>   
>   comparison failed
>   Obtained: -0.06772962212[562](https://github.com/materialsproject/atomate2/actions/runs/8586508423/job/23529051664?pr=789#step:6:563)561
>   Expected: -0.051912 ± 5.2e-06
> FAILED tests/forcefields/test_jobs.py::test_mace_relax_makera[/home/runner/work/atomate2/atomate2/tests/test_data/forcefields/mace/MACE.model-False-True-0.1] - assert -0.06772968173027039 == -0.051912 ± 5.2e-06
>   
>   comparison failed
>   Obtained: -0.06772968173027039
>   Expected: -0.051912 ± 5.2e-06

Reply:
yeah, sorry didn't bother updating the energy in case the changes will be reverted. just commented out the energy assert locally to reach the symmetry check which failed.

pymatgen defaults to [`angle_tolerance=5`](https://github.com/materialsproject/pymatgen/blob/a2ab944bf8e1b90e51ca2b65f853972392177286/pymatgen/symmetry/analyzer.py#L56) like you said while [all the `ase` call sites](https://gitlab.com/ase/ase/-/blob/0ca67f36ff4df1be8790742d005c8337021164f4/ase/spacegroup/symmetrize.py#L93) for `spglib.get_symmetry_dataset` i could find no `angle_tolerance` is passed so the [default and recommended -1](https://spglib.readthedocs.io/en/stable/api/python-api/spglib/spglib.html#spglib.get_symmetry) is used

> Symmetry search tolerance in the unit of angle deg. Normally it is not recommended to use this argument. See a bit more detail at [angle_tolerance](https://spglib.readthedocs.io/en/stable/variable.html#variables-angle-tolerance). If the value is negative, an internally optimized routine is used to judge symmetry.

doesn't seem ideal that `pymatgen` defaults to 5. but would break stuff to change it now...
probably best to just pass `si_structure.get_space_group_info(angle_tolerance=-1)` here and leave pmg as is

Reply:
I think the problem might be the following While the space group staying the same means the symmetries are conserved we can of course arrive at a higher symmetry during optimization which changes the space group but is still the correct results for the fix_symmetry setting. Consequently we need to check if the previous symmetry operations are a subset of the new symmetry operations (This is also how the test of Fix_symmetry in ase is done). 
```
    from pymatgen.io.ase import AseAtomsAdaptor
    from ase.spacegroup.symmetrize import check_symmetry, is_subgroup, FixSymmetry
    si_atoms = AseAtomsAdaptor.get_atoms(si_structure)
    symmetry_ops_init = check_symmetry(si_atoms, symprec=1.0e-3)
    si_atoms_final = AseAtomsAdaptor.get_atoms(output1.output.structure)
    symmetry_ops_final = check_symmetry(si_atoms_final, symprec=1.0e-3)
    if fix_symmetry:
        assert is_subgroup(symmetry_ops_init, symmetry_ops_final)
    else:
        assert not is_subgroup(symmetry_ops_init, symmetry_ops_final)
```

Reply:
I pushed a commit with this but left the rest of the test unchanged so it will still be failing.
If you think it's correct now we can fix the rest.

Reply:
sorry for letting this slip! that makes sense and thanks for pushing a fix. just updated the test assertions to match our changes